docs/guides/Screen-Navigation-Prop.md

```diff
@@ -90,7 +90,29 @@ class ProfileScreen extends React.Component {
 
 ## `goBack` - Close the active screen and move back
 
-Optionally provide a key, which specifies the route to go back from. By default, goBack will close the route that it is called from. If the goal is to go back *anywhere*, without specifying what is getting closed, call `.goBack(null);`
+Optionally provide a key, which specifies the route to go back from. By default, goBack will close the route that it is called from.
+
+*Going back from a specific screen*
+
+Consider the following navigation stack history:
+```... 
+navigation.navigate(SCREEN_KEY_A);
+...
+navigation.navigate(SCREEN_KEY_B);
+...
+navigation.navigate(SCREEN_KEY_C);
+...
+navigation.navigate(SCREEN_KEY_D);
+```
+
```

Comment:
It would be useful to see what is the navigation stack history after each `goBack()` calls.

Reply:
Also I just realized that this part is a bit misleading, because what you pass into navigate is actually just the route name, but what you want to pass into goBack is actually a key generated by the routers, and you can access it with`navigation.state.key`.
So I suggest we simplify the navigation stack as:
ScreenA -> ScreenB -> ScreenC -> ScreenD
And also explain that the key that we pass into goBack is from navigation.state.key.

Reply:
That's a good point @tienf4m. Want to update this PR to reflect that?